.changes/next-release/bugfix-AWSSDKforJavav2-6e2fe73.json

```diff
@@ -0,0 +1,5 @@
+{
+    "category": "AWS SDK for Java v2", 
+    "type": "bugfix", 
+    "description": "Fix the deadlock issue in event streaming operations triggered in an edge case where `request` and `onComplete` are signaled at the same time."
```

Comment:
nit: Can we add the interface names to the methods so it's a little clearer? It could be confusing since `EventStreamResponseHandler` has as similarly named `complete()` method

Reply:
+1 Updated